### PR TITLE
[EJBCLIENT-319] Update affinities on return in NamingEJBClientInterceptor

### PR DESCRIPTION
**Pull request by rachmatowicz**
This PR addresses a problem in affinity processing with the EJB client when the NamingEJBClientInterceptor is used to discover a destination for an invocation.

See the issue for more details: https://issues.jboss.org/browse/EJBCLIENT-319
It is required for the changes in https://issues.jboss.org/browse/WFLY-11489

### CODE REVIEW

File: src/main/java/org/jboss/ejb/client/DiscoveryEJBClientInterceptor.java [hunk 1]
```diff
@@ -122,10 +128,17 @@ public Object handleInvocationResult(final EJBClientInvocationContext context) t
             // set the weak affinity to the location of the session (in case it failed over)
             final Affinity targetAffinity = context.getTargetAffinity();
             if (targetAffinity != null) {
+                if (Logs.INVOCATION.isDebugEnabled()) {
+                    Logs.INVOCATION.debugf("DiscoveryEJBClientInterceptor: calling handleInvocationResult - staeful, clustered case, setting weakAffinity to targetAffinty = %s", targetAffinity);
```
Comment (fl4via) [inline src/main/java/org/jboss/ejb/client/DiscoveryEJBClientInterceptor.java]: Please, fix staeful typo

File: src/main/java/org/jboss/ejb/client/NamingEJBClientInterceptor.java [hunk 2]
```diff
@@ -183,6 +199,9 @@ static boolean setNamingDestination(final AbstractInvocationContext context, fin
                 result.add(check);
             }
         }
+        if (Logs.INVOCATION.isDebugEnabled()) {
+            Logs.INVOCATION.debugf("NamingEJBClientInterceptor: computing preffered URIs: URIs = %s, preferred URIs = %s", uris, result);
```
Comment (fl4via) [inline src/main/java/org/jboss/ejb/client/NamingEJBClientInterceptor.java]: Another typo: preffered instead of preferred

**Comment by rachmatowicz** [comment]
> I'm requesting typo changes, and I'll also do some small verifications in the code before giving my okay

Fixed.

File: src/main/java/org/jboss/ejb/client/DiscoveryEJBClientInterceptor.java (hunk 1 above)
Comment (rachmatowicz) [reply src/main/java/org/jboss/ejb/client/DiscoveryEJBClientInterceptor.java]: Done.

Comment (rachmatowicz) [reply src/main/java/org/jboss/ejb/client/DiscoveryEJBClientInterceptor.java]: Done.

File: src/main/java/org/jboss/ejb/client/NamingEJBClientInterceptor.java (hunk 2 above)
Comment (rachmatowicz) [reply src/main/java/org/jboss/ejb/client/NamingEJBClientInterceptor.java]: Done

**Pull request merged**